### PR TITLE
chore: warn deprecated props

### PR DESCRIPTION
**Pull request by talkor**
https://monday.monday.com/boards/3532714909/pulses/6751163668

### CODE REVIEW

File: packages/core/src/helpers/warnDeprecatedProps.ts
Comment (YossiSaadi) [inline packages/core/src/helpers/warnDeprecatedProps.ts:1]: as it is a hook, maybe the folder name should imply to it?

File: packages/core/src/helpers/warnDeprecatedProps.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useEffect } from "react";
+
+type DeprecatedProps = Record<string, string>;
+
+export function useWarnDeprecatedProps<T extends Record<string, any>>(
```
Comment (YossiSaadi) [inline packages/core/src/helpers/warnDeprecatedProps.ts:5]: you made it generic but haven't used it as generic in any of the usages, is it needed?

File: packages/core/src/helpers/warnDeprecatedProps.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  props: T,
+  deprecatedProps: DeprecatedProps,
```
Comment (YossiSaadi) [inline packages/core/src/helpers/warnDeprecatedProps.ts:7]: maybe alternativeProps? or something similar? it implies for old props, not new one

File: packages/core/src/helpers/warnDeprecatedProps.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  componentName: string
+): void {
+  useEffect(() => {
+    Object.keys(deprecatedProps).forEach(prop => {
+      if (prop in props && props[prop] !== undefined) {
+        console.warn(
+          `${componentName}: '${prop}' prop is deprecated and will be removed in the next major version. Use '${deprecatedProps[prop]}' instead. [monday-ui-react-core]`
```
Comment (YossiSaadi) [inline packages/core/src/helpers/warnDeprecatedProps.ts:14]:
```suggestion
          `[monday-ui-react-core] ${componentName}: '${prop}' prop is deprecated and will be removed in the next major version. Use '${deprecatedProps[prop]}' instead.`
```

Just a suggestion, for you to decide

**Review by YossiSaadi — APPROVED** [review]

**Comment by talkor** [comment]
Decided not to do

**Pull request closed without merging**